src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/Interop/SortKey.Linux.cs

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Text;
+using System.Runtime.InteropServices;
+using System.Security.Principal;
+using System.Runtime.Versioning;
+
+namespace System.DirectoryServices.Protocols
+{
+    [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
+    internal class SortKeyInterop
+    {
+        private string _name;
+        private string _rule;
+        private bool _order;
+
+        public SortKeyInterop()
+        {
+        }
+
+        public SortKeyInterop(string attributeName, string matchingRule, bool reverseOrder)
```

Comment:
I'd define this as a ctor based on the public `SortKey` type.

Further, since the Windows and Linux files differ only on the attribute, I'd break it down into

* SortKeyInterop.cs
  *  Declares a partial struct, has the ctor, fields, whatever else feels necessary.
* SortKeyInterop.Windows.cs
  * Declares the partial struct, applies the StructLayout attribute for Windows
* SortKeyInterop.Unix.cs
  * Declares the partial struct, applies the StructLayout attribute for Unix.

Reply:
Done